test/parallel/test-child-process-execfile.js

```diff
@@ -62,10 +62,23 @@ const execOpts = { encoding: 'utf8', shell: true };
     execFile(process.execPath, [echoFixture, 0], { signal }, check);
   };
 
-  test();
-  ac.abort();
   // Verify that it still works the same way now that the signal is aborted.
   test();
+  ac.abort();
+}
+
+{
+  // Verify that does not spawn a child if already aborted
+  const ac = new AbortController();
+  const { signal } = ac;
+  ac.abort();
+
+  try {
+    execFile(process.execPath, [echoFixture, 0], { signal });
+  } catch (err) {
+    assert.strictEqual(err.code, 'ABORT_ERR');
+    assert.strictEqual(err.name, 'AbortError');
+  }
```

Comment:
```suggestion
  assert.throws(
    () => execFile(process.execPath, [echoFixture, 0], { signal }),
    {
      name: 'AbortError',
      code: 'ABORT_ERR',
    }
  );
```

Reply:
This approach sadly it's not possible as the error is handled within the callback following the pattern `(err, result) => {}`.
What do you think about this suggestion?
```suggestion
  const check = common.mustCall((err) => {
    assert.strictEqual(err.code, 'ABORT_ERR');
    assert.strictEqual(err.name, 'AbortError');
    assert.strictEqual(err.signal, undefined);
  });
  execFile(process.execPath, [echoFixture, 0], { signal }, check);
```

Reply:
Yeah that looks good to me!